client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -2035,10 +1973,30 @@
          */
         this.close = function() {
             self.item = null;
-            self.readonly = null;
-            superdeskFlags.flags.authoring = false;
-            $location.search('edit', null);
-            $location.search('view', null);
+            self.action = null;
+            saveState();
+        };
+
+        /**
+         * Kill an item
+         *
+         * @param {Object} item
+         */
+        this.kill = function(item) {
+            self.item = item._id || null;
+            self.action = 'kill';
+            saveState();
+        };
+
+        /**
+         * Correct an item
+         *
+         * @param {Object} item
+         */
+        this.correct = function(item) {
+            self.item = item._id || null;
+            self.action = 'correct';
+            saveState();
```

Comment:
Seems like this action-setting logic repeats itself in several methods, with the only difference being the action name. It probably makes sense to refactor it to a common helper function?